[WIP] Get userId from GraphQl using Auth0 credentials

nickeyvee opened the pull request:
### Description
When authenticating with Auth0 and receiving profile information we immediately query our database and store our userId in our root component state.

#### The issue
When signing in for the first time and trying to view all of the user's boards all of the data is not available yet. You can see this behavior by immediately clicking on `boards` when you land on the Home view.

![screenshot 174](https://user-images.githubusercontent.com/20284107/42792898-761d8cd6-8945-11e8-83e2-479a2dfdb160.png)

Click `home` and then re-click `boards` to see the rest of the data.

![screenshot 175](https://user-images.githubusercontent.com/20284107/42792906-85cf6ffa-8945-11e8-9081-dc06fc1da43f.png)

nickeyvee committed:
:sparkles: (App.js, Home.js, BoardsHome.js) Gets userId from GraphQl using Auth0 key when App is re…

nickeyvee committed:
:recycle: (BoardsHome.js) Retrieves all data in one request instead of two.

albertoescala reviewed (approved):
Good Job @nickeyvee! 🥇 💯

nickeyvee committed:
:sparkles: (BoardsHome.js, BoardsHome.styles, LeftSidebar.js) Pass boardId as a url parameter.

LeftSidebar now recieves userId as props for making db queries.

nickeyvee commented:
## Update 
The **board id** is passed as a url parameter when you click on a board.

<img width="349" alt="params" src="https://user-images.githubusercontent.com/20284107/42796461-4539a3da-8958-11e8-8a10-3ac049adfb56.PNG">

nickeyvee committed:
:sparkles: (App.js, Board.js) Query database with board Id obtained from url params.

boardId is no longer hard-coded and obtained via url params when a board is clicked on.

albertoescala commented on src/components/Home/components/BoardsHome/BoardsHome.styles.js:
```diff
@@ -41,9 +42,29 @@ const CardWrapper = styled.div`
+  color: white;
+
+  &:hover {
+    filter: brightness(80%);
```
Interesting 👍
